### ISSUE TITLE
Json Provider should not flatten structure

### ISSUE THREAD

**Issue by erikwiborg** [cfg4j/cfg4j#162]
hi, I'm playing around with cfg4j and as we're using JSON in our current setup I'd like to continue using it. However, much of the point of using json is lost if you can't use arrays and objects (and arrays of objects) - particularly when the flattening produces strange results - e.g. 
```"key": [{"innerKey": "value"}] ``` fattens to just "key": "value"

is there a particular reason why structures are flattened? The consumer of the configuration can be expected to know about the general structure of the configuration, no?

**Comment by norbertpotocki** [comment]
Hey @erikwiborg - this is probably a bug - we just added JSON support in the last release. Can you attach a few JSON examples that you use for me to run?

**Comment by erikwiborg** [comment]
sure, this is the one i've been using: 
```
{
"moduleName": "name",
"seeds": ["one", "two", "three"],
"databases": [
  {
    "host": "localhost", 
    "port": 27017    
  }]
}```

thanks!

**Comment by ruslansennov** [comment]
@norbertpotocki I've seen this problem when work with PR, but I do not know how to implement it without a radical alteration of the architecture of all project. Hope you can do it :smile:

**Comment by norbertpotocki** [comment]
Yeah, that's because the Java property files don't support arrays. I guess we'll have to implement our own extension to the format. Let's see what I can do about this.

**Comment by ruslansennov** [comment]
Also it's unclear how we should store array of objects in consul...

**Comment by kofrasa** [comment]
You could enforce a convention in a java properties file to represent arrays or objects. The example above can be written as

```
moduleName=name
seeds.$0=one
seeds.$1=two
seeds.$2=3
databases.$0.host=localhost
databases.$0.port=27017
```

Am prefixing the number with ```$``` to safe guard against user input that make use of numeric keys. You can also provide a ```Cf4gProperties``` class which extends ```Properties``` to facilitates retrieving the nested values.

```java
cf4gProps.getCfgProperty("databases", 0, "host"); // return value for "database.$0.host"
cf4gProps.getProperty("database.$0.host"); // equivalent to above
```
